src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -539,6 +595,34 @@ def __getitem__(self, index_or_slice: SupportsIndex | slice) -> float | FloatArr
         """
         return self._buffer.__getitem__(index_or_slice)
 
+    def _covered_time_range(self) -> timedelta:
+        """Return the time range that is covered by the oldest and newest valid samples.
+
+        Returns:
+            The time range between the oldest and newest valid samples or 0 if
+                there are is no time range covered.
+        """
+        if not self.oldest_timestamp:
+            return timedelta(0)
+
+        assert (
+            self.newest_timestamp is not None
+        ), "Newest timestamp cannot be None here."
+        return self.newest_timestamp - self.oldest_timestamp + self._sampling_period
+
+    @property
+    def count_covered(self) -> int:
```

Comment:
I know I'm annoying, but I find it confusing that the method name has a verb but it is a property. I would either

1. Make this a method (remove `@property`)
2. Since it's just a simple math that is done, it could stay a property, but then I would rename it as `num_covered`, `covered_count` or a similar noun.

Also, I think I prefer not to use *count* in this case, because at least for me counting conveys the action of a linear loop (or incrementing a number every time something happens), counting 1, 2, 3, 4 and this is not really counting but just calculating.

If we keep it a method, then I think is fine to call it `count` because it can be an abstraction, specially if we want to keep the naming *symmetry* with `count_valid()`.

I think probably the simplest would be to just remove the `@property`...